test/e2e/framework/service.go

```diff
@@ -0,0 +1,84 @@
+package framework
+
+import (
+	"time"
+
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+type ServiceType int
+
+const (
+	InvalidServicePodType ServiceType = iota
+	NginxService
+)
+
+type ServiceConfig struct {
+	Type    ServiceType
+	Cluster ClusterIndex
+	// TODO: namespace, once https://github.com/submariner-io/submariner/pull/141 is merged
+}
+type Service struct {
+	Service            *corev1.Service
+	Config             *ServiceConfig
+	TerminationCode    int32
+	TerminationMessage string
+	framework          *Framework
+}
+
+func (f *Framework) NewService(config *ServiceConfig) *Service {
+
+	// check if all necessary details are provided
+	Expect(config.Type).ShouldNot(Equal(InvalidServicePodType))
+
+	service := &Service{Config: config, framework: f, TerminationCode: -1}
+
+	switch config.Type {
+	case NginxService:
+		service.buildNginxService()
+	}
+	return service
+}
```

Comment:
I think this can be simplified to:

```
func (f *Framework) NewNginxService(cluster ClusterIndex) *corev1.Service {
...
}
```

That would eliminate the `ServiceType`, `ServiceConfig`, and `Service` types.

Reply:
?It will be useful when we add a new service right?

Reply:
Done.

Reply:
Similar to the Deployment code,  I suggest inlining `buildNginxService` with a `func (s *Service) create(...)` method. Also add a `func (s *Service) Delete()` method.

Reply:
Removed the struct.